### ISSUE TITLE
4.0.1 still breaks all the menus!!!

### ISSUE THREAD

**Issue by Paul-Tahoe** [progressivetech/net.ourpowerbase.sumfields#45]
Now the error is:  *15745 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Class 'CRM_Myextension_ExtensionUtil' not found in /usr/share/nginx/xxx.xxx.org/sites/default/files/civicrm/custom_ext/net.ourpowerbase.sumfields/sumfields.php on line 30" 

I think the problem is this.

You added 

require_once 'sumfields.civix.php';
use CRM_Myextension_ExtensionUtil as E;

As far as I can tell Myextension doesn't exists.

It probably should be 

use CRM_Sumfields_ExtensionUtil as E;

Which seems to work.

**Comment by colemanw** [comment]
@Paul-Tahoe I don't see any reference to the string "Myextension" in this repository.

**Comment by jmcclelland** [comment]
I see it:

```
--- a/sumfields.php
+++ b/sumfields.php
@@ -1,7 +1,7 @@
 <?php
 
 require_once 'sumfields.civix.php';
-use CRM_Myextension_ExtensionUtil as E;
+use CRM_Sumfields_ExtensionUtil as E;
 
 /**
  * Implementation of hook_civicrm_config
```
There is something definitely wrong with my testing infrastructure - I don't seem to be properly clearing my caches when I test my updates.

**Comment by jmcclelland** [comment]
I'm really sorry everyone for the series of mistakes in this latest release. 

I just tagged v4.0.2. I still see some behavior I would like to change on a fresh install (it prompts you to  upgrade the schema even though it was just installed from scratch). However, in the interest of fixing this latest bug I am just pushing out a version with fix in my last comment.

**Comment by fatihates** [comment]
I did the CiviCRM 5.10.1 update. Menu's gone. The solution is not healthy if this problem occurs in every update.

**Comment by Paul-Tahoe** [comment]
Jamie, for some reason version 4.0.2 isn't available from within civi.  So the latest upgrade shows as 4.0.1.  I'm not sure where civi actually pulls code from when you hit the upgrade button.   I see on the https://civicrm.org/extensions/summary-fields it only shows up to 4.0.1.   Of course, if one pulls it from github (e.g. git clone ...) it pulls the correct stuff.

**Comment by colemanw** [comment]
Releases go out every 24 hours. It's published now.

**Comment by Paul-Tahoe** [comment]
It's too late now for this one but for the future is there a way for a developer to un-release a version that has been found to be bad?

**Comment by colemanw** [comment]
Yes the tag can be removed from github and the release node can be deleted from civicrm.org.

**Comment by jmcclelland** [comment]
This issue now be resolved with 4.0.2
